source/agora/consensus/EnrollmentManager.d

```diff
@@ -390,7 +390,7 @@ public class EnrollmentManager
         // Generate the random seed to use
         auto cache = PreImageCache(PreImageCycle.NumberOfCycles, cycle_length);
         assert(offset < cache.length);
-        cache.reset(hashMulti(kp.v, "consensus.preimages", offset));
+        cache.reset(hashMulti(kp.v, "consensus.preimages", cast(uint)offset));
```

Comment:
The commit message should say what the actual bug was. For example:

```
Fixed using the wrong type (ulong instead of uint) when creating the preimage hash.
```

Reply:
We should just change the parameter type maybe